src/DataTable/index.jsx

```diff
@@ -69,64 +71,31 @@ function DataTable({
     // Stringifying the data gives a quick way of checking deep equality
   }, [fetchData, JSON.stringify(instance.state)]);
 
-  const {
-    getTableProps,
-    getTableBodyProps,
-    headerGroups,
-    rows,
-    prepareRow,
-  } = instance;
-
-  const filterNames = instance.state.filters ? instance.state.filters.map((filter) => filter.id) : [];
-  const resetAllFilters = instance.setAllFilters ? () => instance.setAllFilters([]) : null;
-  const pageSize = instance.page ? instance.page.length : rows.length;
+  const propsForChildren = { itemCount, ...rest };
+  const childrenWithExtraProps = React.Children.map(
+    children, child => React.cloneElement(child, propsForChildren),
+  );
 
   return (
-    <div className="pgn__data-table-wrapper">
-      <TableControlBar
-        isSelectable={isSelectable}
-        selectedFlatRows={instance.selectedFlatRows}
-        toggleAllRowsSelected={instance.toggleAllRowsSelected}
-        isFilterable={isFilterable}
-        filterNames={filterNames}
-        pageSize={pageSize}
-        itemCount={itemCount}
-        bulkActions={bulkActions}
-        columns={instance.columns}
-        rows={instance.flatRows}
-        resetAllFilters={resetAllFilters}
-        numBreakoutFilters={numBreakoutFilters}
-      />
-      {rows.length > 0 && (
-        <Table
-          getTableProps={getTableProps}
-          getTableBodyProps={getTableBodyProps}
-          headerGroups={headerGroups}
-          /* the page contains only the rows in it, as opposed to all rows.
-            it is only available when the table is paginated */
-          rows={instance.page ? instance.page : rows}
-          prepareRow={prepareRow}
-        />
-      )}
-      {rows.length <= 0 && <EmptyTableComponent />}
-      <TableFooter
-        itemCount={itemCount}
-        pageSize={pageSize}
-        isPaginated={isPaginated}
-        previousPage={instance.previousPage}
-        nextPage={instance.nextPage}
-        canNextPage={instance.canNextPage}
-        canPreviousPage={instance.canPreviousPage}
-        pageIndex={instance.state.pageIndex}
-        pageCount={instance.pageCount}
-      />
-    </div>
+    <DataTableContext.Provider value={instance}>
```

Comment:
Where does `instance` come from?

Reply:
It comes from `useTable` (from `react-table`). The child components all deal with it not being there gracefully. (I mean, as gracefully as possible)